stats/release_queue.php

```diff
@@ -11,143 +11,263 @@
 
 require_login();
 
-if ($ordinary_users_can_see_queue_settings) {
-    $user_can_see_queue_settings = true;
-} else {
-    // Only privileged users can see queue settings
-    $user_can_see_queue_settings = user_is_a_sitemanager() || user_is_proj_facilitator();
-}
-
 $round_id = get_enumerated_param($_GET, 'round_id', null, array_keys($Round_for_round_id_), true);
 $name = @$_GET['name'];
+$enabled_only = get_integer_param($_GET, 'enabled_only', 1, 0, 1);
+$unheld_only = get_integer_param($_GET, 'unheld_only', 0, 0, 1);
+
+$title = _("Release Queues");
+output_header($title, NO_STATSBAR);
+echo "<h1>" . html_safe($title) . "</h1>\n";
 
 if (is_null($round_id)) {
-    $title = _("Release Queues");
-    output_header($title, NO_STATSBAR);
-    echo "<h1>" . html_safe($title) . "</h1>\n";
+    _show_available_rounds();
+} else {
+    $round = get_Round_for_round_id($round_id);
+
+    if (isset($name)) {
+        _show_queue_details($round, $name, $unheld_only);
+    } else {
+        _show_round_queues($round, $enabled_only);
+    }
+}
+
+//-----------------------------------------------------------------------------
+
+function _show_available_rounds()
+{
+    global $Round_for_round_id_;
 
     echo _("Each round has its own set of release queues."), "\n";
     echo _("Please select the round that you're interested in:"), "\n";
     echo "<ul>\n";
     foreach (array_keys($Round_for_round_id_) as $round_id) {
-        echo "<li><a href='release_queue.php?round_id=$round_id'>$round_id</a></li>\n";
+        echo "<li><a href='?round_id=$round_id'>$round_id</a></li>\n";
     }
     echo "</ul>\n";
-    return;
 }
 
-$round = get_Round_for_round_id($round_id);
+function _show_round_queues($round, $enabled_only)
+{
+    global $code_url;
 
-if (!isset($name)) {
-    $title = sprintf(_("Release Queues for Round '%s'"), $round_id);
-    output_header($title, NO_STATSBAR);
-    echo "<h1>" . html_safe($title) . "</h1>\n";
+    echo "<p>";
+    echo "<a href='?'>" . _("Back to queue round selection") ."</a> | ";
```

Comment:
Do we want to html_safe these strings similar to the title and other loc strings?

Reply:
Yes and done. It's so annoying that we don't have some framework that handles all of this for us.